sql/src/main/java/io/crate/analyze/DataTypeAnalyzer.java

```diff
@@ -33,6 +33,8 @@
 
 public class DataTypeAnalyzer extends DefaultTraversalVisitor<DataType, Void> {
 
+    public final static DataTypeAnalyzer INSTANCE = new DataTypeAnalyzer();
```

Comment:
please also add a private ctor to enforce using the singleton. 

Instead of exposing the INSTANCE it would also be possible to have a static `DataType convert(Expression expr)` method